ecommerce/api.py

```diff
@@ -734,11 +751,20 @@ def complete_order(order):
     if order_coupon_ids:
         set_coupons_to_redeemed(order.purchaser.email, order_coupon_ids)
 
-    # clear the basket
     with transaction.atomic():
-        BasketItem.objects.filter(basket__user=order.purchaser).delete()
-        CourseRunSelection.objects.filter(basket__user=order.purchaser).delete()
-        CouponSelection.objects.filter(basket__user=order.purchaser).delete()
+        baskets = Basket.objects.filter(
+            id__in=(
+                Basket.objects.select_for_update(skip_locked=True).filter(
+                    user=order.purchaser
+                )
+            )
+        )
+        basket_ids = baskets.values_list("id", flat=True)
+        log.info("Found %d baskets to delete", len(basket_ids))
+
+        # clear the basket
+        clear_baskets(basket_ids)
+        baskets.delete()
```

Comment:
Why don't we move this to the clear_baskets method? Can we name it clear_and_delete baskets?